src/Categories/NaturalTransformation/NaturalIsomorphism.agda

```diff
@@ -71,6 +71,31 @@ record NaturalIsomorphism {C : Category o ℓ e}
       }
     }
 
+-- Like IsIso for morphisms, a natural transformation may be an isomorphism
+-- if there exists some inverse to it.
+-- To be consistent, there's also an IsNI for natural isomorphisms
+record IsNI {C : Category o ℓ e}
+            {D : Category o′ ℓ′ e′}
+            {F G : Functor C D}
+            (F⇒G : NaturalTransformation F G) : Set (o ⊔ ℓ ⊔ ℓ′ ⊔ e′) where
+  field
+    inv : NaturalTransformation G F
+
+  module F⇒G = NaturalTransformation F⇒G
```

Comment:
As these 2 modules are only used once each, and only for 1 field, I don't think they are worth defining. [I know this is the pattern in other parts of the library, but I'm moving away from it as it causes a fair amount of bloat in the actual interfaces.]

You could keep them if you 1) make them private, and 2) add `using (η)` to them. Then they are quite thin, and that's fine.